python/cuml/internals/memory_utils.py

```diff
@@ -449,7 +415,7 @@ def set_global_output_type(output_type):
         raise ValueError('Parameter output_type must be one of "numpy", '
                          '"cupy", cudf", "numba", "input" or None')
 
-    cuml.global_settings.output_type = output_type
+    global_settings.output_type = output_type
 
 
 @contextlib.contextmanager
```

Comment:
Just setting the output type? No, I wouldn't think so. Doing the validity checks beforehand? Possibly. We can profile, but for this PR it may make more sense to just holistically evaluate our benchmark suite and then dig into any hotspots that reveals.

Reply:
Ok, I was more thinking of a scenario where a user uses the CM in some inner loop for type conversion and is then affected by the `contextlib.contextmanager` slowness.

Reply:
Oh! I'm sorry, I thought you were just referring to the actual setting of the output type variable. Yes, I'm definitely a fan of ripping out `contextlib.contextmanager` wherever we have it, and this could indeed be a place where performance would suffer. I'll do a quick conversion to a class.